Makefile

```diff
@@ -67,9 +66,24 @@ setup_paas_env: set_cf_target
 build: require_env_stub get_git_status ## Create & tag a new docker image
 	docker build -t $(APP_NAME)-$(env_stub) --build-arg GIT_COMMIT_SHA=$(git_commit_sha) --build-arg GIT_BRANCH=$(git_branch) .
 
-push: require_env_stub ## push the Docker image to Docker Hub
-	docker tag $(APP_NAME)-$(env_stub) $(REMOTE_DOCKER_IMAGE_NAME)-$(env_stub)
-	docker push $(REMOTE_DOCKER_IMAGE_NAME)-$(env_stub)
+push: require_env_stub timestamp-latest ## push the Docker image to Docker Hub
+	docker tag $(APP_NAME)-$(env_stub) $(REMOTE_DOCKER_IMAGE_NAME)-$(env_stub):latest
+	docker push $(REMOTE_DOCKER_IMAGE_NAME)-$(env_stub):latest
+
+timestamp-latest: require_env_stub
+	docker pull $(REMOTE_DOCKER_IMAGE_NAME)-$(env_stub):latest
+	$(eval export timestamp_tag=replaced-at-$(shell (date -u +%Y%m%d-%H%M%S)))
+	docker tag  $(REMOTE_DOCKER_IMAGE_NAME)-$(env_stub):latest $(REMOTE_DOCKER_IMAGE_NAME)-$(env_stub):$(timestamp_tag)
+	docker push $(REMOTE_DOCKER_IMAGE_NAME)-$(env_stub):$(timestamp_tag)
+
+remote-docker-tags: require_env_stub
+	@curl -L -s 'https://registry.hub.docker.com/v2/repositories/$(REMOTE_DOCKER_IMAGE_NAME)-$(env_stub)/tags?page_size=1024' |jq -r '."results"[]["name"]'
```

Comment:
This assumes `jq` is installed on the system, which isn't a given on all Linux & mac systems? I think we need to list it as a pre-requisite somewhere otherwise this would fail?

Reply:
good spot, I've added it to the list of pre-requisites